idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/intentions/SafeAccessToIfThenIntention.kt

```diff
@@ -57,11 +64,24 @@ class SafeAccessToIfThenIntention : SelfTargetingRangeIntention<KtSafeQualifiedE
             isAssignment = true
         }
 
+        val replacedReference = (ifExpression.then as? KtQualifiedExpression)?.callExpression?.let {
+            val replaceSingleLineLetIntention = ReplaceSingleLineLetIntention()
+            if (replaceSingleLineLetIntention.isApplicableTo(it)) {
+                replaceSingleLineLetIntention.applyTo(it, editor)
+                val descriptor = (ifExpression.condition as? KtBinaryExpression)?.left?.resolveToCall()?.resultingDescriptor
+                ifExpression.then?.collectDescendantsOfType<KtNameReferenceExpression>()?.firstOrNull {
+                    it.resolveToCall()?.resultingDescriptor == descriptor
+                }
+            } else {
+                null
+            }
+        }
+
```

Comment:
I'm not quite sure about this code. First of all, the `replaceReference` is required to be calculated only in one branch, see the only usage. Second, I'd rename `let` parameter here, otherwise it's difficult to understand the code. And at last, this search among `KtNameReferenceExpression`s looks suspicious. Too many of them can exist in general case and I suspect (may be I'm wrong) this can give false positives.